proto/uber/cadence/api/v1/service_domain.proto

```diff
@@ -158,3 +160,49 @@ message ListDomainsResponse {
   repeated Domain domains = 1;
   bytes next_page_token = 2;
 }
+
+message ListFailoverHistoryRequest {
+  // ListFailoverHistoryRequestFilters specifies the filters to apply to the request.
+  // If not provided all failover events will be returned.
+  ListFailoverHistoryRequestFilters filters = 1;
```

Comment:
We were talking about letting web service do the filtering. Maybe we can remove this and add it later if we want to do backend filtering?